subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -41,7 +43,10 @@ static void dir_list_object_encode(struct bt_gatt_ots_object *obj,
 	net_buf_simple_add_le48(net_buf, obj->id);
 
 	/* Name length */
-	net_buf_simple_add_u8(net_buf, strlen(obj->metadata.name));
+	obj_name_len = strlen(obj->metadata.name);
+	__ASSERT(obj_name_len >= 0 && obj_name_len <= 120,
+		 "Dir list object len is too small %u", len);
```

Comment:
```suggestion
		 "Dir list object len is incorrect %u", len);
```

It can be too large.

The first condition seems always true:

```obj_name_len >= 0 ```

Reply:
The first condition should have been `obj_name_len > 0` as empty strings are not allowed for the names.